### DISCUSSION TITLE
Curso: Fullstack .Net - Identity - Login

### DISCUSSION THREAD

**Discussion by danilobicas-unoeste** [balta-io/forum#647] [Tira Dúvidas] [answered]
Olá.
consegui fazer o Register customizado pq já estou implementando umas coisas diferentes para adequar para um projeto pessoal. Cadastrou o user na tabela normalmente. Mas ao tentar fazer o login (chamando o v1/identity mesmo) ele me retorna 401 - Unauthorized. Aí me bateu uma dúvida. O objeto "user" usado no identity está lá no projeto webapi/Models e o user que eu uso no projeto blazor(frontend) é referenciado no projeto CORE/Models(LoginRequest)..... são objetos diferentes. Perdi alguma coisa nas aulas? Qual seria o problema? muito obrigado.

**Comment by andrebaltieri** [comment, marked as answer]
@danilobicas-unoeste como vai?

Não sei se entendi muito bem ou se são duas perguntas separadas... a primeira é do 401... Sobre isto preciso de mais informações... pode ser problemas de usuário/senha inválidos.

Sobre a sua segunda dúvida.. sim, são objetos separados, que fiam em projetos separados... Eles podem ficar no Core, mas o Core não pode referenciar o ASP.NET Identity... se não ferir esta regra, tudo bem!

**Reply by danilobicas-unoeste** [reply]
Olá. Na verdade a minha dúvida é se o erro 401 seria devido a usar dois objetos (user) um no front e outro no backend poderia ocasionar este erro. Aí, então minha dúvida é se perdi algum detalhe pq no front eu parto com um objeto (LoginRequest)... mas este objeto não deveria ser mapeado para o "user do identity" lá no servidor/api??? Como o identity recebe um objeto que não é herdado do "user" do identity?
Não sei se me fiz entender.....

Agadeço.

**Reply by andrebaltieri** [reply]
Boa tarde, @danilobicas-unoeste!

Na verdade a minha dúvida é se o erro 401 seria devido a usar dois objetos (user) um no front e outro no backend poderia ocasionar este erro. 
 - Não! O erro 401 é quando não está autorizado!

Aí, então minha dúvida é se perdi algum detalhe pq no front eu parto com um objeto (LoginRequest)... mas este objeto não deveria ser mapeado para o "user do identity" lá no servidor/api??? 
 - Estou um pouco perdido nesta sua parte da dúvida... consegue postar um pedaço do código que está e o que quer fazer?


Como o identity recebe um objeto que não é herdado do "user" do identity?
 - Me passa o código que está querendo executar e mais detalhes da sua dúvida... talvez um passo-a-passo.

**Comment by DiegoTelis** [comment]
Achei o problema parecido com o meu.   Deixa eu tentar ajudar, a me ajudar rsrss
Creio o que ele citou sobre as classes User seria as criadas no Dima.Api-Models-User  e a Dima.Core-Models-Identity-User.  Se causaria uma confusão sobre o retorno dela ..   

Adicionar mais coisas aqui, imaginando que deve ser o mesmo caso meu.  
Ele passa do Login do Handler retornando 200,  mas na parte de retornar o usuario nos (metodos) CookieAuthenticationStateProvider > GetAuthenticationStateAsync > GetUser  ..   retornou esse erro 401.. 

![image](https://github.com/user-attachments/assets/97b7da58-6794-44fc-9fe5-0ffe23e4e7da)

Engraçado que pelo Swager ou retorna 200 quando o cookie esta criado,  e 404 quando não..  Aqui no login chega a criar ele também...
Ou seja.. buguei a mente aqui rss

![image](https://github.com/user-attachments/assets/1d13efa8-7a48-4185-9fb2-bf7160fa9159)

**Reply by DiegoTelis** [reply]
.... uééee....   
Eu rodava o dotnet run para API e debugava o Web..  Só que esse Web estava rodando no HttpS..  Pensei tipo "ahh tanto faz ... não?!" rss    

Resumindo botei pra rodar tudo via http  ajustei o Endereço do front pro CORS e foi....      O porque .. Sei lá x.x
